tests/rptest/tests/topic_limits_test.py

```diff
@@ -0,0 +1,88 @@
+# Copyright 2025 Redpanda Data, Inc.
+
+#
+# Use of this software is governed by the Business Source License
+# included in the file licenses/BSL.md
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0
+
+from rptest.tests.redpanda_test import RedpandaTest
+from rptest.services.cluster import cluster
+from rptest.clients.types import TopicSpec
+from rptest.clients.rpk import RpkTool
+from rptest.clients.kafka_cli_tools import KafkaCliTools
+
+
+class TopicLimitsTest(RedpandaTest):
```

Comment:
Can you look at the equivalent test for partition limits, and ensure this covers the same ground, e.g., error checking, etc.